### PR TITLE
List unecessary env vars for source instrumentations

### PR DESCRIPTION
**Pull request by pellared**
## Why

Acting as .NET profiler my be unnecessary and may descrease the applications performance.

## What

List the env vars that are not needed when bytecode instrumentation is not required.

## Tests

None (sic!)

## Checklist

- [ ] ~~`CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- [ ] ~~New features are covered by tests.~~

### CODE REVIEW

**Review by dszmigielski — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -139,6 +139,22 @@ OTEL_SERVICE_NAME=my-service
...
+COR_PROFILER_PATH_32
+COR_PROFILER_PATH_64
+CORECLR_ENABLE_PROFILING
+CORECLR_PROFILER
+CORECLR_PROFILER_PATH
+CORECLR_PROFILER_PATH_32
+CORECLR_PROFILER_PATH_64
+OTEL_DOTNET_AUTO_INTEGRATIONS_FILE
+```
+
```
Comment (theletterf) [inline docs/README.md:157]:
```suggestion
If you don't need [bytecode instrumentations](config.md#instrumentations),
you can unset or remove the following environment variables:

```env
COR_ENABLE_PROFILING
COR_PROFILER
COR_PROFILER_PATH_32
COR_PROFILER_PATH_64
CORECLR_ENABLE_PROFILING
CORECLR_PROFILER
CORECLR_PROFILER_PATH
CORECLR_PROFILER_PATH_32
CORECLR_PROFILER_PATH_64
OTEL_DOTNET_AUTO_INTEGRATIONS_FILE
```

```

**Review by theletterf — APPROVED** [review]
Approved with some suggestions

**Review by pjanotti — APPROVED** [review]
Yeah, good idea listing those, but, having a test would be nice to back it up :)

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Comment by pellared** [comment]
> Yeah, good idea listing those, but, having a test would be nice to back it up :)

It is. There is `RunTestApplication(agent.Port, enableClrProfiler: !IsCoreClr());` in most of tests for source instrumentations.

**Pull request merged**